tests/e2e/sizing_policy_test.go

```diff
@@ -0,0 +1,174 @@
+/*
+Copyright 2024 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"slices"
+	"strconv"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	virtv2 "github.com/deckhouse/virtualization/api/core/v1alpha2"
+	kc "github.com/deckhouse/virtualization/tests/e2e/kubectl"
+)
+
+func ValidateVirtualMachineByClass(virtualMachineClass *virtv2.VirtualMachineClass, virtualMachine *virtv2.VirtualMachine) {
+	var sizingPolicy virtv2.SizingPolicy
+	for _, p := range virtualMachineClass.Spec.SizingPolicies {
+		if virtualMachine.Spec.CPU.Cores >= p.Cores.Min && virtualMachine.Spec.CPU.Cores <= p.Cores.Max {
+			sizingPolicy = *p.DeepCopy()
+			break
+		}
+	}
+
+	checkMinMemory := virtualMachine.Spec.Memory.Size.Value() >= sizingPolicy.Memory.Min.Value()
+	checkMaxMemory := virtualMachine.Spec.Memory.Size.Value() <= sizingPolicy.Memory.Max.Value()
+	checkMemory := checkMinMemory && checkMaxMemory
+	Expect(checkMemory).To(BeTrue(), fmt.Errorf("memory size outside of possible interval '%v - %v': %v", sizingPolicy.Memory.Min, sizingPolicy.Memory.Max, virtualMachine.Spec.Memory.Size))
+
+	coreFraction, err := strconv.Atoi(strings.ReplaceAll(virtualMachine.Spec.CPU.CoreFraction, "%", ""))
+	Expect(err).NotTo(HaveOccurred(), "cannot convert CoreFraction value to integer: %s", err)
+	checkCoreFraction := slices.Contains(sizingPolicy.CoreFractions, virtv2.CoreFractionValue(coreFraction))
+	Expect(checkCoreFraction).To(BeTrue(), fmt.Errorf("sizing policy core fraction list does not contain value from spec: %s\n%v", virtualMachine.Spec.CPU.CoreFraction, sizingPolicy.CoreFractions))
+}
+
+var _ = Describe("Sizing policy", Ordered, ContinueOnFailure, func() {
+	var (
+		vmNotValidSizingPolicy   string
+		phaseByVolumeBindingMode string
+		vmClassDiscovery         string
+		notExistingVmClass       = map[string]string{"vm": "not-existing-vmclass"}
+		existingVmClass          = map[string]string{"vm": "existing-vmclass"}
+		testcaseLabel            = map[string]string{"testcase": "sizing-policy"}
+	)
+
+	Context("Environment preparing", func() {
+		vmNotValidSizingPolicy = fmt.Sprintf("%s-vm-%s", namePrefix, notExistingVmClass["vm"])
+		vmClassDiscovery = fmt.Sprintf("%s-discovery", namePrefix)
+		switch conf.StorageClass.VolumeBindingMode {
+		case "Immediate":
```

Comment:
May be exclude this logic for common e2e tests place?

Reply:
moved it to init func in tests_suite_test.go